lib/application_insights/channel/contracts/data.rb

```diff
@@ -1,48 +1,9 @@
 require_relative 'json_serializable'
 
-module ApplicationInsights
-  module Channel
-    module Contracts
-      # Data contract class for type Data.
-      class Data < JsonSerializable
-        # Initializes a new instance of the Data class.
-        def initialize(options={})
-          defaults = {
-            'baseType' => nil,
-            'baseData' => nil
-          }
-          values = {
-            'baseData' => nil
-          }
-          super defaults, values, options
-        end
-        
-        # Gets the base_type property.
-        def base_type
-          @values.fetch('baseType') { 
-            @values['baseType'] = nil
-          }
-        end
-        
-        # Sets the base_type property.
-        def base_type=(value)
-          if value == @defaults['baseType']
-            @values.delete 'baseType' if @values.key? 'baseType'
-          else
-            @values['baseType'] = value
-          end
-        end
-        
-        # Gets the base_data property.
-        def base_data
-          @values['baseData']
-        end
-        
-        # Sets the base_data property.
-        def base_data=(value)
-          @values['baseData'] = value
-        end
-      end
-    end
+module ApplicationInsights::Channel::Contracts
+  class Data
+    include JsonSerializable
+
+    attributes :baseType, :baseData
```

Comment:
The _hash mapping_ is the same for every instance of the same Contract class. Given that, it should be defined only once instead on every new instance. Also we are repeating the same keys twice, one for `attr_accessor` and one for the _hash mapping_. That is why I suggest to group them in a simpler interface, which in the background uses `attr_accessor` method.

On the other hand, we can go with something like this
```ruby
attr_accessor :account_acquisition_date, :account_id, :user_agent, :id

json_mapping account_acquisition_date: 'ai.user.accountAcquisitionDate',
               account_id: 'ai.user.accountId',
               user_agent: 'ai.user.userAgent',
               id: 'ai.user.id'  
```
But I don't like the idea of **repeated keys**